core/src/services/gdrive/backend.rs

```diff
@@ -0,0 +1,241 @@
+    root: String,
+    access_token: String,
+    client: HttpClient,
+}
+
+impl GoogleDriveBackend {
+    pub(crate) fn new(root: String, access_token: String, http_client: HttpClient) -> Self {
+        GoogleDriveBackend {
+            root,
+            access_token,
+            client: http_client,
+        }
+    }
+}
+
+impl Debug for GoogleDriveBackend {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut de = f.debug_struct("GoolgeDriveBackend");
+        de.field("root", &self.root);
+        de.finish()
+    }
+}
+
+#[async_trait]
+impl Accessor for GoogleDriveBackend {
+    type Reader = IncomingAsyncBody;
+    type BlockingReader = ();
+    type Writer = GoogleDriveWriter;
+    type BlockingWriter = ();
+    type Pager = ();
+    type BlockingPager = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut ma = AccessorInfo::default();
+        ma.set_scheme(crate::Scheme::GDrive)
+            .set_root(&self.root)
+            .set_capability(Capability {
+                read: true,
+                write: true,
+                delete: true,
+                ..Default::default()
+            });
+
+        ma
+    }
+
+    async fn read(&self, path: &str, _args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let resp = self.gdrive_get(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                let meta = parse_into_metadata(path, resp.headers())?;
+                Ok((RpRead::with_metadata(meta), resp.into_body()))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
+        if args.content_length().is_none() {
+            return Err(Error::new(
+                ErrorKind::Unsupported,
+                "write without content length is not supported",
+            ));
+        }
+
+        Ok((
+            RpWrite::default(),
+            GoogleDriveWriter::new(self.clone(), args, String::from(path)),
+        ))
+    }
+
+    async fn delete(&self, path: &str, _: OpDelete) -> Result<RpDelete> {
+        let resp = self.gdrive_delete(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::NO_CONTENT => Ok(RpDelete::default()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+}
+
+impl GoogleDriveBackend {
+    async fn get_abs_root_id(&self) -> String {
+        let mut req = Request::get("https://www.googleapis.com/drive/v3/files/root");
+        let auth_header_content = format!("Bearer {}", self.access_token);
+        req = req.header(header::AUTHORIZATION, auth_header_content);
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)
+            .unwrap();
+
+        let resp = self.client.send(req).await.unwrap();
+
+        let body_value: serde_json::Value =
```

Comment:
Please use storage typed json instead.

Reply:
There are different response format for different api, opendal gdrive service support involve several api, should we define different struct for the diffenent api response?

Reply:
Yes.